compiler/semantic/expr.go

```diff
@@ -415,13 +415,26 @@ func semAssignment(scope *Scope, a ast.Assignment, summarize bool) (dag.Assignme
 			return dag.Assignment{}, fmt.Errorf("lhs of assigment expression: %w", err)
 		}
 	} else if call, ok := a.RHS.(*ast.Call); ok {
-		// If LHS is nil and the call is every() make the LHS field ts since
-		// field ts assumed with every.
-		name := call.Name
-		if name == "every" {
-			name = "ts"
+		var path []string
+		switch call.Name {
+		case "every":
+			// If LHS is nil and the call is every() make the LHS field ts since
+			// field ts assumed with every.
+			path = []string{"ts"}
+		case "quiet":
+			if len(call.Args) > 0 {
+				if p, ok := rhs.(*dag.Call).Args[0].(*dag.This); ok {
+					path = p.Path
+				} else {
+					return dag.Assignment{}, fmt.Errorf("implied quiet variable must be a path")
```

Comment:
Shouldn't this be the same error as the default case?

Reply:
I'm not quite sure I follow. If do something like `cut quiet(a.b[0])` this is valid selector but we can't imply a variable name here.